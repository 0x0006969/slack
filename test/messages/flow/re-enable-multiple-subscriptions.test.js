const ReEnableMultipleSubscriptions = require('../../../lib/messages/flow/re-enable-multiple-subscriptions');

describe('ReEnableMultipleSubscriptions message rendering', () => {
  test('works', async () => {
    const subscribedRepositories = [
      {
        full_name: 'kubernetes/kubernetes',
      },
      {
        full_name: 'facebook/react',
      },
      {
        full_name: 'django/django',
      },
      {
        full_name: 'microsoft/typescript',
      },
    ];
    const message = new ReEnableMultipleSubscriptions(
      subscribedRepositories,
      'U12345',
      'disconnected their GitHub account',
    );
    expect(message.getAttachment()).toMatchSnapshot();
  });
});
