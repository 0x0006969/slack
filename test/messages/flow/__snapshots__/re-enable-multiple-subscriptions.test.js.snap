// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ReEnableMultipleSubscriptions message rendering works 1`] = `
Object {
  "color": "#24292f",
  "mrkdwn_in": Array [
    "text",
  ],
  "text": "Subscriptions to 4 repositories have been disabled, because <@U12345>, who originally set them up, has disconnected their GitHub account.
Use the following slash commands to re-enable the subscriptions:
/github subscribe kubernetes/kubernetes
/github subscribe facebook/react
/github subscribe django/django
/github subscribe microsoft/typescript",
}
`;
